app/uk/gov/hmrc/gform/ApplicationLoader.scala

```diff
@@ -181,52 +182,56 @@ class ApplicationModule(context: Context)
   val jsonCrypto =
     SymmetricCryptoFactory.aesCryptoFromConfig(baseConfigKey = "json.encryption", configModule.typesafeConfig)
 
+  val mongoCacheRepository = createMongoCacheRepository("forms")
   val formMongoCache = new FormMongoCache(
-    new MongoCacheRepository[String](
-      mongoModule.mongoComponent,
-      "forms",
-      true,
-      configModule.appConfig.formExpiryDays.days,
-      new CurrentTimestampSupport(),
-      SimpleCacheId
-    ) {
-      override def ensureIndexes: Future[Seq[String]] = {
-        val formExpiry = configModule.appConfig.formExpiryDays.days.toMillis
-        val createdFormExpiry = configModule.appConfig.createdFormExpiryDays.days.toMillis
-        val submittedExpiry = configModule.appConfig.submittedFormExpiryHours.hours.toMillis
-        val indexes = Seq(
-          IndexModel(
-            Indexes.ascending("modifiedDetails.createdAt"),
-            IndexOptions()
-              .background(false)
-              .name("createdAtIndex")
-              .expireAfter(createdFormExpiry, TimeUnit.MILLISECONDS)
-          ),
-          IndexModel(
-            Indexes.ascending("modifiedDetails.lastUpdated"),
-            IndexOptions()
-              .background(false)
-              .name("lastUpdatedIndex")
-              .expireAfter(formExpiry, TimeUnit.MILLISECONDS)
-          ),
-          IndexModel(
-            Indexes.ascending("submitDetails.createdAt"),
-            IndexOptions()
-              .background(false)
-              .name("submittedIndex")
-              .expireAfter(submittedExpiry, TimeUnit.MILLISECONDS)
-          )
-        )
-        MongoUtils.ensureIndexes(this.collection, indexes, true)
-      }
-    },
```

Comment:
We should not be reusing this for snapshot data.

Reply:
We don't reuse `form` repository for snapshot data. It is just refactoring. Production code is not being changed. We just use the production instance of the `form` repository in the TestOnlyFormService to copy/restore to/from snapshot repository.